analyzers/src/SonarAnalyzer.CSharp/Rules/ForLoopCounterChanged.cs

```diff
@@ -43,19 +44,24 @@ public sealed class ForLoopCounterChanged : SonarDiagnosticAnalyzer
         private sealed class SideEffectExpression
         {
             public IImmutableList<SyntaxKind> Kinds { get; set; }
-            public Func<SyntaxNode, SyntaxNode> AffectedExpression { get; set; }
+            public Func<SyntaxNode, ImmutableArray<SyntaxNode>> AffectedExpressions { get; set; }
         }
 
         private static readonly IImmutableList<SideEffectExpression> SideEffectExpressions = ImmutableArray.Create(
             new SideEffectExpression
             {
                 Kinds = ImmutableArray.Create(SyntaxKind.PreIncrementExpression, SyntaxKind.PreDecrementExpression),
-                AffectedExpression = node => ((PrefixUnaryExpressionSyntax)node).Operand
+                AffectedExpressions = node => ImmutableArray.Create<SyntaxNode>(((PrefixUnaryExpressionSyntax)node).Operand)
             },
             new SideEffectExpression
             {
                 Kinds = ImmutableArray.Create(SyntaxKind.PostIncrementExpression, SyntaxKind.PostDecrementExpression),
-                AffectedExpression = node => ((PostfixUnaryExpressionSyntax)node).Operand
+                AffectedExpressions = node => ImmutableArray.Create<SyntaxNode>(((PostfixUnaryExpressionSyntax)node).Operand)
+            },
+            new SideEffectExpression
+            {
+                Kinds = ImmutableArray.Create(SyntaxKindEx.TupleExpression),
+                AffectedExpressions = node => ImmutableArray.Create<SyntaxNode>(((TupleExpressionSyntaxWrapper)node).Arguments.Select(x => x.Expression).ToArray())
```

Comment:
I will replace `Arguments.Select(x => x.Expression).ToArray()` with the extension method from this [PR](https://github.com/SonarSource/sonar-dotnet/pull/5770).